seurat-export-cellbrowser.R

```diff
@@ -0,0 +1,308 @@
+    c("-o", "--output-directory"),
+    action = "store",
+    default = NA,
+    type = 'character',
+    help = "File name in which to store serialized R object of type 'Seurat'.'"
+  ),
+  make_option(
+    c("-n", "--study-name"),
+    action= "store",
+    default = "Seurat_study",
+    type="character",
+    help="Study name to be displayed in CellBrowser"
+  ),
+  make_option(
+    c("-m", "--markers-file"),
+    default = NULL,
+    type="character",
+    help="Path to markers file computed by Seurat"
+  )
+)
+
+opt <- wsc_parse_args(option_list, mandatory = c('input_object_file', 'output_directory'))
+opt$study_name = gsub(" ","_", opt$study_name) # study names cannot have spaces.
+
+# Check parameter values
+
+if ( ! file.exists(opt$input_object_file)){
+  stop((paste('File', opt$input_object_file, 'does not exist')))
+}
+
+# Input from serialized R object
+
+seurat_object <- readRDS(opt$input_object_file)
+
+
+
+# the following is copied from Seurat3's utilities.R and generics.R
+# it was modified to work with Seurat2. At some point we probably should merge the two code bases.
+# the markers "---" are used to mark the part that is used by the command line tool cbImportSeurat2
+
+# ---
+#' Export Seurat object for UCSC cell browser
+#'
+#' @param object Seurat object
+#' @param dir output directory path
+#' @param dataset.name name of the dataset
+#' @param meta.fields vector of metadata fields to export
+#' @param meta.fields.names list that defines metadata field names
+#'                          after the export. Should map metadata
+#'                          column name to export name
+#' @param embeddings vector of embedding names to export
+#' @param markers.file path to file with marker genes
+#' @param cluster.field name of the metadata field containing cell cluster
+#' @param cb.dir in which dir to create UCSC cell browser content root
+#' @param port \emph{experimental} on which port to run UCSC cell browser after export
+#'
+#' @export
+#'
+#' @importFrom reticulate py_module_available
+#' @importFrom reticulate import
+#'
+#' @examples
+#' \dontrun{
+#' ExportToCellbrowser(object = pbmc_small, dataset.name = "PBMC", dir = "out")
+#' }
+#'
+ExportToCellbrowser <- function(
+  object,
+  dir,
+  dataset.name,
+  meta.fields = NULL,
+  meta.fields.names = NULL,
+  embeddings = c("tsne", "pca", "umap"),
+  markers.file = NULL,
+  cluster.field = NULL,
+  port = NULL,
+  cb.dir = NULL,
+  markers.n = 100,
+  skip.expr.matrix = FALSE,
+  skip.markers = FALSE,
+  all.meta = FALSE
+) {
+  if (!require("Seurat",character.only = TRUE)) {
+    stop("This script requires that Seurat (V2 or V3) is installed")
+  }
+  message("Seurat Version installed: ", packageVersion("Seurat"))
+  message("Object was created with Seurat version ", object@version)
+  
+  if (substr(object@version, 1, 1)!='2') {
+    stop("can only process Seurat2 objects, version of rds is ", object@version)
+  }
+  
+  #idents <- FetchData(object,c("ident"))$ident;
```

Comment:
Assume you'll be removing commented lines before releasing this ;-)

Reply:
I just copy pasted this from Seurat's I think (or CellBrowser). We can do, but honestly I don't feel inclined to do any changes within this function because I hope to remove it once we use it from the upstream. I think that some of those comments show what would be the call like on Seurat 3 apparently.